powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v2/ActivationServiceBehavior.java

```diff
@@ -204,6 +204,14 @@ public PrepareActivationResponse prepareActivation(String activationIdShort, Str
             throw localizationProvider.buildExceptionForCode(ServiceError.ACTIVATION_EXPIRED);
         }
 
+        // Make sure activation code has 23 characters
+        if (activation.getActivationCode().length() != 23) {
```

Comment:
You are right that we generate the activation code. I included the check because the value is read from database (which means it is externally modifiable) and substring is used later on. I think it is ok to remove this validation.

Reply:
Do not remove it, just isolate it in some private validation method...

Reply:
Ok, sounds good.